### PR TITLE
cli-plugins/manager: minor cleanups and refactoring

### PR DESCRIPTION
**Pull request by thaJeztah**
### cli-plugins/manager: ListPlugins: pass context to error-group

This error-group was added in 89583b92b74ec3c2dfe770b0c01437b1e373c1aa, but
passed a context.TODO because the function didn't have a context as argument.

However, it does get the root-command passed, which holds the context, so
we can pass that.

### cli-plugins/manager: add test for empty / non-existing plugin dirs

Verify that listPluginCandidates returns an empty result if nothing was
found.

### cli-plugins/manager: ListPlugins: return early if no candidates

Skip the other logic, which includes listing all commands provided; if
there's no plugin-candidates, those steps won't be needed.

### cli-plugins/manager: getPluginDirs: remove redundant error-return

This function returned an error (if any) from [config.Path]. However, the
only situation in which an error could be returned was if the given path
to append to `config.Dir` was outside of the config directory. This can
only happen if the path to append would try to traverse directories (e.g.,
passing `../../cli-plugins`).

Given that we're passing a hard-coded value, that would not be the case,
so we can simplify the code to join the path directly, and don't have to
handle errors.

[config.Path]: https://github.com/docker/cli/blob/2d74733942be353bc7730c8722ae2414f368b732/cli/config/config.go#L100-L107

**- Human readable description for the release notes**

```markdown changelog


```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Ah! Was wondering indeed if there would be code not liking a `nil` slice to be returned; I can change it to return an empty slice, but we should probably fix the calling code as well to handle it gracefully;

```
 > [test 2/2] RUN --mount=type=bind,target=.,rw     --mount=type=cache,target=/root/.cache     --mount=type=cache,target=/go/pkg/mod     gotestsum -- -coverprofile=/tmp/coverage.txt $(go list ./... | grep -vE '/vendor/|/e2e/'):
62.37 github.com/docker/cli/cli-plugins/manager.ListPlugins({0x1008460?, 0xc0001718c8?}, 0xc00018bc00)
62.37 	/go/src/github.com/docker/cli/cli-plugins/manager/manager.go:148 +0x1ab
62.37 github.com/docker/cli/cli-plugins/manager.TestListPluginsIsSorted(0xc000394340)
62.37 	/go/src/github.com/docker/cli/cli-plugins/manager/manager_test.go:151 +0x407
62.37 testing.tRunner(0xc000394340, 0xf14bb0)
62.37 	/usr/local/go/src/testing/testing.go:1690 +0xf4
62.37 created by testing.(*T).Run in goroutine 1
62.37 	/usr/local/go/src/testing/testing.go:1743 +0x390
```

**Comment by thaJeztah** [comment]
Oh! Ignore me; It's the context that's `nil` - interesting; I recall we once had a discussion about that, but Cobra itself was expected to set this automatically 🤔 

```
=== RUN   TestListPluginsIsSorted
--- FAIL: TestListPluginsIsSorted (0.00s)
panic: cannot create context from nil parent [recovered]
	panic: cannot create context from nil parent
```

File: cli-plugins/manager/manager.go [hunk 1]
```diff
@@ -145,16 +138,20 @@ func getPlugin(name string, pluginDirs []string, rootcmd *cobra.Command) (*Plugi
...
 
 	var plugins []Plugin
 	var mu sync.Mutex
-	eg, _ := errgroup.WithContext(context.TODO())
+	ctx := rootcmd.Context()
+	if ctx == nil {
+		// Fallback, mostly for tests that pass a bare cobra.command
+		ctx = context.Background()
+	}
+	eg, _ := errgroup.WithContext(ctx)
```
Comment (Benehiko) [inline cli-plugins/manager/manager.go:154]: maybe we should instead force the tests to set a context?

Comment (thaJeztah) [reply cli-plugins/manager/manager.go:154]: Yeah, I originally updated the test, but given that this function is exported, I could not 100% exclude the possibility for other code to call this with a command that has not (yet?) has a context set.

Without a context, this produces a panic at runtime, which was a bit risky, so ultimately I chose for making this "best effort", and keep the test as-is to verify that things work even if no context is present on the command.

Comment (Benehiko) [reply cli-plugins/manager/manager.go:154]: Ah i see, alright then let's get this one merged.

**Review by Benehiko — APPROVED** [review]
